Sophos Central - The new parameter to the integration.

paoloGuagnano opened the pull request:
…o filter by the Product the alarms to fetch.

<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
The new parameter `Fetch Product` has been added to the integration to filter by the Product the alarms to fetch.

## Must have
- [ ] Tests
- [ ] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=30100) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=30100) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=30100) it.</sub>

sapirshuker commented:
> Thank you for your contribution. Your generosity and caring are unrivaled! Make sure to register your contribution by filling the [Contribution Registration](https://forms.gle/XDfxU4E61ZwEESSMA) form, so our content wizard @ilappe will know the proposed changes are ready to be reviewed. For your convenience, here is a [link](https://xsoar.pan.dev/docs/contributing/sla) to the contributions SLAs document.

Hey @paoloGuagnano,
 thank you for your contribution.
I will review your PR shorty, in the meantime, please fill in the following [Contribution Registration](https://forms.gle/XDfxU4E61ZwEESSMA) form.

sapirshuker commented on Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.yml:
```diff
@@ -32,6 +32,23 @@ configuration:
   - high
   type: 16
   required: false
+- additionalinfo: Choose one or more products to fetch. If none is chosen, all products will be returned.
```
Is the product being fetched or the alarm? if it the alarm so please add more detailed description.

sapirshuker reviewed (changes requested):
Hey,thanks a lot for this contribution, It is highly appreciated.
I have reviewed it but unfortunately it has some issues.
I have posted some comments in the review.
In addition, there are a build failures, please [review](https://app.circleci.com/pipelines/github/demisto/content/308248/workflows/f8b8f29b-f0e5-494e-abad-06a412e3908e/jobs/723437) and fix them.
I'm available for any of your questions here or on slack.

sapirshuker commented:
Hey, @paoloGuagnano 
You still have some problems with the build 
Please review and fix it.
Do you need any help with the pull request?
Please feel free to reach out to me here or on Slack.

sapirshuker commented:
Hey, @paoloGuagnano , we haven’t heard from you in a while.
Do you need any help with the pull request?
Please feel free to reach out.
Thanks again for contributing to our repo, hope to hear from you soon.

sapirshuker commented on Packs/SophosCentral/Integrations/SophosCentral/README.md:
```diff
@@ -13,6 +13,7 @@ The unified console for managing Sophos products.
 | isFetch | Fetch incidents | False |
 | fetch_severity | Fetch Severity | False |
 | fetch_category | Fetch Category | False |
+| fetch_product | Fetch Product | False |
```
Please generate only the relevant commands/parameters without delete the command examples.

sapirshuker commented on Packs/SophosCentral/Integrations/SophosCentral/SophosCentral_test.py:
```diff
@@ -1,4330 +1,4331 @@
-    EntryType,
-    set_integration_context,
-    get_integration_context,
-)
-from SophosCentral import Client
```
Please revert the deletion and add only the relevant changes

paoloGuagnano replied:
Hi sapirshuker, I don't quite understand the nature of this error. Would you be so kind as to explain the error to me better? Thank you

sapirshuker commented on Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.yml:
```diff
@@ -32,9 +32,29 @@ configuration:
   - high
   type: 16
   required: false
+- additionalinfo: Choose one or more products to fetch. If none is chosen, all alarms generated by all products will be returned.
```
Please update the docker image

sapirshuker reviewed (changes requested):
Great Job! 
Please see my comments
We are waiting for the partner approval for this change.

sapirshuker commented:
Hey @paoloGuagnano,
You are doing great! 
We are waiting for a partner approval.
You can also try to solve the errors in the [build](https://app.circleci.com/pipelines/github/demisto/content/311222/workflows/7d787b48-acf3-4130-b37d-c047dd0f59cf/jobs/725505).
I'm available for any of your questions here or on slack.
Thanks again.

sapirshuker commented:
Hey, @paoloGuagnano 
A meeting was secluded with the panther 
In the meantime, Please see my review
You can also try to solve the errors in the [build](https://app.circleci.com/pipelines/github/demisto/content/311222/workflows/7d787b48-acf3-4130-b37d-c047dd0f59cf/jobs/725505).
I'm available for any of your questions here or on slack.
Thanks again.

sapirshuker commented:
Hey, @paoloGuagnano Sure, I will write to you an explanation about this error.
But first, I saw that you added 5000+ files, into the PR in this 
https://github.com/demisto/content/pull/30100/commits/35a3d4906e48b6665b8a79a54ce68cc3d599c2f4 commit. I believe this was not done on purpose; if so, please revert the change.
Do you know how to revert a commit? 
I'm available for any of your questions here or on slack.
Thanks again.

paoloGuagnano commented:
Hi sapirshuker, 
I actually made a commit error. I corrected the error by returning to commit "72ecdd1" and added the change to the test. Thank you for your support.

sapirshuker commented:
Hey, @paoloGuagnano 
Hey, I can still see a bunch of changed files. Do you know how to revert those changes?
Please revert.
I'm available for any of your questions here or on slack.
Thanks again.
<img width="374" alt="image" src="https://github.com/demisto/content/assets/49246861/e2ecfce1-3b82-4d97-93aa-3c1c8c12fa02">

paoloGuagnano commented:
hi, I used the "git revert" command returning to the commit "Update the docker image - 72ecdd1". Can you check if it's OK?
![image](https://github.com/demisto/content/assets/127855096/3f6bab92-397b-4cea-94c0-663375dbb371)

sapirshuker commented:
> hi, I used the "git revert" command returning to the commit "Update the docker image - [72ecdd1](https://github.com/demisto/content/commit/72ecdd1befd48d554184d2311250f0c860484e10)". Can you check if it's OK? ![image](https://user-images.githubusercontent.com/127855096/278292852-3f6bab92-397b-4cea-94c0-663375dbb371.png)

Hey, @paoloGuagnano 
This is correct, but you chose the wrong commit, I changed it for you. Please don't hesitate to reach out if you have any issues, As soon as the build process is complete, I will let you know if there are any errors and how to resolve them.
Thanks again.

paoloGuagnano commented:
Hi sapirshuker, 
thanks so much for the support.

sapirshuker commented:
Hey, @paoloGuagnano 
I would like to arrange the changes you made to the test file. 
Is the [03cf2334](https://github.com/demisto/content/pull/30100/commits/9c178e58fc9c59bfa37fcbbd81f6dc3803cf2334) only change you made to the test file?

paoloGuagnano commented:
Hi sapirshuker,
Correct, these[03cf2334](https://github.com/demisto/content/pull/30100/commits/9c178e58fc9c59bfa37fcbbd81f6dc3803cf2334)  are the changes I made in the test file

sapirshuker commented:
Hey, @paoloGuagnano 
We are still waiting for partner approval for the contribution.
In the meantime, you can resolve the README issue. 
To do that, please run the `demisto-sdk generate-docs` command (use the -e flag for using a commad_examples file) - for more information please see: https://xsoar.pan.dev/docs/concepts/demisto-sdk#generate-docs.
I'm available for any of your questions here or on slack.
Thanks again.

paoloGuagnano commented:
hi [sapirshuker](https://github.com/sapirshuker),
I have already tried to generate the docs using this command 
<demisto-sdk generate-docs -e Packs/SophosCentral/Integrations/SophosCentral/command_examples.txt -i Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.yml">

Maybe I made some mistake?
I did not modify the "command_examples.txt" file as I did not add commands.

Thanks for the support.

sapirshuker commented:
Hey @paoloGuagnano 
We are still waiting for partner approval for the contribution.
My suggestion is that I will help you the generate the README after the contribution will be approved by the partner.
Please note that you will required to add test if the contribution will be approved. 
Thanks

sapirshuker commented:
Hey @paoloGuagnano
We are still waiting for partner approval for the contribution.

paoloGuagnano commented:
hi https://github.com/sapirshuker,
ok, thanks for letting me know

edik24 commented:
Hi @paoloGuagnano, we are having issues contacting the Partner and getting approval.
So currently the status is that it is being handed internally by our Business Development team directly with the partner.
We will update how to proceed with that.
But let's keep the PR open.

paoloGuagnano commented:
hi [edik24](https://github.com/edik24), 
is there any news? 
Thank you

edik24 commented:
> hi [edik24](https://github.com/edik24), is there any news? Thank you

Hi,
Not yet.
But we're on that. I'll hope I'll have an update early next week

paoloGuagnano commented:
OK,
 thank you.

sapirshuker commented:
Hey, @paoloGuagnano 
Great news! 
We decided to move this pack to be in XSOAR support level. 
We have an issue with our repo, so your PR is messy right now We will fix it, so don't panic about all the files you have in your PR. We will fix it soon.
We will review your PR soon. 
Thank you for your patience and contribution.

sapirshuker commented:
Your contribution was impacted by changes in the git history of the content repo, and you see many file changes that are not related to your PR. Our team is currently working on a solution, and we hope that it will be resolved soon.

sapirshuker commented:
Last Thursday, December 20th, there was an update in the content repo. This is why your PR looks like there were a lot of commits and files changed.
To fix this issue please do the following:

1.Save all of your changes locally or in a new branch.
2. Enter your forked repository page in GitHub.
3. At the top of the page, you will probably see something like this, please ensure you are in the master branch.
image
![image](https://github.com/demisto/content/assets/49246861/7fdc505d-4e49-4928-b0da-dca7cfe5e181)
4. Click on “Sync fork” button on the top right side of the page.
5. The following message will appear (This branch has conflicts that must be resolved).
image
![image](https://github.com/demisto/content/assets/49246861/e28b5e33-de0b-43d0-8bf3-5252ec9ba356)

6. Click on the button “Discard commits”. Note that for this action to work, you should have permission to do a force push in the forked repo. If you don’t have one, ask the forked repo Owner to do that.
7. Enter your branch and repeat the same steps (4-6).

Check your PR, and see if your changes have been saved. If yes - you finished the update process.
If your PR does not show your changes, open a new PR with the changes that you saved at the beginning of the process.
In case you opened a new PR please close the old one, and update your reviewer with the new PR link.
Please paste on the new open PR the URL to your previous PR so our team member can continue the review from the same place.

Thank you very much for your contribution and your understanding of the situation.
Have a nice holiday season!

paoloGuagnano commented:
hi [sapir](https://github.com/sapirshuker),
sorry for the delay but I was on holiday.
I hope I made the changes correctly.
I am waiting for your response.
Thank you

sapirshuker commented:
> Last Thursday, December 20th, there was an update in the content repo. This is why your PR looks like there were a lot of commits and files changed. To fix this issue please do the following:
> 
> 1.Save all of your changes locally or in a new branch. 2. Enter your forked repository page in GitHub. 3. At the top of the page, you will probably see something like this, please ensure you are in the master branch. image ![image](https://private-user-images.githubusercontent.com/49246861/292731953-7fdc505d-4e49-4928-b0da-dca7cfe5e181.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUyMTkyNTgsIm5iZiI6MTcwNTIxODk1OCwicGF0aCI6Ii80OTI0Njg2MS8yOTI3MzE5NTMtN2ZkYzUwNWQtNGU0OS00OTI4LWIwZGEtZGNhN2NmZTVlMTgxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE0VDA3NTU1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg3ZTY3NTQxOTBlZWJmN2NiNzdkNGM1NGM3NzgxYjlkMmMyNmQwMjY4NWZlNTI0MTk1OTlhOWI3YmMwZjVlMmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.98MWXzjEEZxwomoigSn5IGPbagiWD5DRNTJRE-sy1YE) 4. Click on “Sync fork” button on the top right side of the page. 5. The following message will appear (This branch has conflicts that must be resolved). image ![image](https://private-user-images.githubusercontent.com/49246861/292732001-e28b5e33-de0b-43d0-8bf3-5252ec9ba356.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUyMTkyNTgsIm5iZiI6MTcwNTIxODk1OCwicGF0aCI6Ii80OTI0Njg2MS8yOTI3MzIwMDEtZTI4YjVlMzMtZGUwYi00M2QwLThiZjMtNTI1MmVjOWJhMzU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE0VDA3NTU1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM1YjM0MzZmNjllNjY2ZThmOWU0MmFmZTBmYzEzN2Y2MTk5MDUzZTI0NjdlMzZiNDM1ZTUxZmIxZWMxNzQ4YmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8SRB-luu-QwixMn8GZDdWneoIoViHe0CY3KLxPgpXmM)
> 
> 6. Click on the button “Discard commits”. Note that for this action to work, you should have permission to do a force push in the forked repo. If you don’t have one, ask the forked repo Owner to do that.
> 7. Enter your branch and repeat the same steps (4-6).
> 
> Check your PR, and see if your changes have been saved. If yes - you finished the update process. If your PR does not show your changes, open a new PR with the changes that you saved at the beginning of the process. In case you opened a new PR please close the old one, and update your reviewer with the new PR link. Please paste on the new open PR the URL to your previous PR so our team member can continue the review from the same place.
> 
> Thank you very much for your contribution and your understanding of the situation. Have a nice holiday season!

Hey @paoloGuagnano 
Thanks! 
Please open a new PR with your changes as mentioned.

sapirshuker commented:
Hey @paoloGuagnano 
Do you need any help with the PR? 
I'm available for any of your questions here or on slack.
Thanks again.

sapirshuker commented:
Hey @paoloGuagnano 
Thank you for your contribution. 
I have not received a response from you in the last two weeks.
Please feel free to open it when you are available to continue. 
Thanks again.
